File: check_2023_names.js
const fs = require('fs');
const https = require('https');

const API_KEY = "4eb0b3baf194555ef46565fa9dc2d35d";
const LEAGUES = [292, 293];
const SEASON = 2022;

const swiftContent = fs.readFileSync('KScout/Services/KoreanTranslationService.swift', 'utf8');
const keys = new Set([...swiftContent.matchAll(/"([^"]+)":/g)].map(m => m[1].toLowerCase()));

function fetchData(url) {
    return new Promise((resolve, reject) => {
        https.get(url, { headers: { 'x-apisports-key': API_KEY } }, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => resolve(JSON.parse(data)));
        }).on('error', reject);
    });
}

async function run() {
    const allNames = new Set();
    for (const league of LEAGUES) {
        for (const endpoint of ['topscorers', 'topassists']) {
            const url = `https://v3.football.api-sports.io/players/${endpoint}?season=${SEASON}&league=${league}`;
            try {
                const res = await fetchData(url);
                if (res.response) {
                    res.response.forEach(item => allNames.add(item.player.name));
                }
            } catch (e) {}
        }
    }

    const missing = [];
    for (const name of Array.from(allNames).sort()) {
        const lowerName = name.trim().toLowerCase();
        let found = false;
        if (keys.has(lowerName)) found = true;
        else {
            for (const k of keys) {
                if (k.length > 3 && lowerName.includes(k)) {
                    found = true;
                    break;
                }
            }
        }
        if (!found) missing.push(name);
    }

    console.log("MISSING NAMES IN DICTIONARY:");
    missing.forEach(n => console.log(n));
}
run();

File: check_2023_names.py
import re
import urllib.request
import json
import ssl

with open('KScout/Services/KoreanTranslationService.swift', 'r', encoding='utf-8') as f:
    swift_content = f.read()

keys = re.findall(r'"([^"]+)":', swift_content)
keys = set([k.lower() for k in keys])

API_KEY = "4eb0b3baf194555ef46565fa9dc2d35d"
LEAGUES = [292, 293]
SEASON = 2023

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

def fetch_data(url):
    req = urllib.request.Request(url, headers={'x-apisports-key': API_KEY})
    with urllib.request.urlopen(req, context=ctx) as response:
        return json.loads(response.read().decode())

all_names = set()

for league in LEAGUES:
    for endpoint in ['topscorers', 'topassists']:
        url = f"https://v3.football.api-sports.io/players/{endpoint}?season={SEASON}&league={league}"
        try:
            res = fetch_data(url)
            if 'response' in res:
                for item in res['response']:
                    all_names.add(item['player']['name'])
        except Exception as e:
            pass

missing = []
for name in sorted(list(all_names)):
    lower_name = name.strip().lower()
    found = False
    if lower_name in keys:
        found = True
    else:
        for k in keys:
            if len(k) > 3 and k in lower_name:
                found = True
                break
    if not found:
        missing.append(name)

print("MISSING NAMES IN DICTIONARY:")
for name in missing:
    print(name)
